src/Employer/Employer.Web/Controllers/VacancyManageController.cs

```diff
@@ -29,36 +34,53 @@ public async Task<IActionResult> DisplayVacancy(VacancyRouteModel vrm)
                 return HandleRedirectOfDraftVacancy(vacancy);
             }
 
-            var m = await _orchestrator.GetVacancyDisplayViewModelAsync(vacancy);
-            return View(m.ViewName, m.ViewModel);
-        }
+            if (vacancy.Status != VacancyStatus.Live)
+            {
+                return RedirectToRoute(RouteNames.DisplayVacancy_Get);
+            }
 
-        [HttpGet("view", Name = RouteNames.DisplayFullVacancy_Get)]
-        public async Task<IActionResult> DisplayFullVacancy(VacancyRouteModel vrm)
-        {
-            var vacancy = await _orchestrator.GetVacancy(vrm);
+            DateTime parsedClosingDate = DateTime.MinValue;
```

Comment:
Is there a more succinct/easier to read way of writing this?

https://github.com/SkillsFundingAgency/das-recruit/blob/7362f28c5cf1e3d460f751fb099107b2d372a0db/src/Employer/Employer.Web/Controllers/VacancyManageController.cs#L42-L58

Reply:
I struggled to think of a way to to make this look more readable, partly because of the return redirects. If you have any suggestion?